core/domain/acl_decorators.py

```diff
@@ -126,3 +167,320 @@ def test_can_view_stats(self, exploration_id, **kwargs):
+        else:
+            raise self.UnauthorizedUserException(
+                'User %s does not have permissions to delete exploration %s' %
+                (self.user_id, exploration_id))
+
+    return test_can_delete
+
+
+def can_suggest_changes_to_exploration(handler):
+    """"Decorator to check whether a user can make suggestions to an
+    exploration.
+    """
+    def test_can_suggest(self, exploration_id, **kwargs):
+        if (
+                role_services.ACTION_SUGGEST_CHANGES_TO_EXPLORATION in
+                self.actions and user_services.has_fully_registered(
+                    self.user_id)):
+            return handler(self, exploration_id, **kwargs)
+        else:
+            raise self.UnauthorizedUserException(
+                'You do not have credentials to give suggestions to ' +
+                'exploration.')
+
+    return test_can_suggest
+
+
+def can_manage_suggestions_on_exploration(handler):
+    """Decorator to check whether a user can act upon the suggestions made
+    by learners."""
+
+    def test_can_manage(self, exploration_id, **kwargs):
+        if not self.user_id:
+            self.redirect(current_user_services.create_login_url(
+                self.request.uri))
+            return
+        exploration_rights = rights_manager.get_exploration_rights(
+            exploration_id)
+
+        if exploration_rights is None:
+            raise self.PageNotFoundException
+
+        if check_exploration_editable(
+                self.user_id, self.actions, exploration_rights):
+            return handler(self, exploration_id, **kwargs)
+        else:
+            raise self.UnauthorizedUserException(
+                'You do not have the credentials to manage suggestions for ' +
+                'this exploration.')
+
+    return test_can_manage
+
+
+def can_publish_exploration(handler):
+    """Decorator to check whether user can publish exploration."""
+
+    def test_can_publish(self, exploration_id, **kwargs):
+        exploration_rights = rights_manager.get_exploration_rights(
+            exploration_id)
+
+        if exploration_rights is None:
+            raise self.PageNotFoundException
+
+        if exploration_rights.cloned_from:
+            raise self.UnauthorizedUserException(
+                'You do not have credentials to publish this exploration.')
+
+        if role_services.ACTION_PUBLISH_ANY_EXPLORATION in self.actions:
+            return handler(self, exploration_id, **kwargs)
+
+        if exploration_rights.status == rights_manager.ACTIVITY_STATUS_PRIVATE:
+            if role_services.ACTION_PUBLISH_OWNED_EXPLORATION in self.actions:
+                if self.user_id in exploration_rights.owner_ids:
+                    return handler(self, exploration_id, **kwargs)
+
+        if exploration_rights.status == rights_manager.ACTIVITY_STATUS_PUBLIC:
+            if role_services.ACTION_PUBLISH_PUBLIC_EXPLORATION in self.actions:
```

Comment:
ah, yes, this is my doubt also. See below:
```
    def can_publicize(self, activity_type, activity_id):
        """Returns whether this user is allowed to publicize the given
        activity.

        Args:
            activity_type: str. The type of activity. Possible values:
                feconf.ACTIVITY_TYPE_EXPLORATION
                feconf.ACTIVITY_TYPE_COLLECTION
            activity_id: str. ID of the activity.

        Returns:
            bool. Whether the user has a right to publicize the given activity.
        """
        activity_rights = _get_activity_rights(activity_type, activity_id)
        if activity_rights is None:
            return False

        if activity_rights.status != ACTIVITY_STATUS_PUBLIC:
            return False
        return self.is_moderator()
```
here function returns true when user is moderator and exploration is public.